libyul/optimiser/CallGraphGenerator.cpp

```diff
@@ -30,6 +32,38 @@ using namespace solidity;
 using namespace solidity::yul;
 using namespace solidity::util;
 
+std::set<YulString> CallGraph::recursiveFunctions() const
+{
+	// TODO: This algorithm is non-optimal.
+	struct CycleFinder {
```

Comment:
Move this to anonymous namespace outside the function?

Reply:
I'm moving it, but why?
Usually I like to keep the scopes as clean as possible :-). That's why I usually also go for local lambda recursions, until you tell me to extract them :-)...

Reply:
Yeah, that is obviously a matter of style, but my problem with excess use of local lambdas and local types is that you need to make another mental step of deciding whether something is a statement or a definition and if it is a statement, whether it is executed right away or just captured inside some lambda function or function inside a struct and executed at some later time. Especially for lambda functions, extracting them into an anonymous namespaces makes it clear that they cannot capture without having to look at the capturing statement.